lib/nxt_state_machine/state_machine.rb

```diff
@@ -0,0 +1,197 @@
+module NxtStateMachine
+  class StateMachine
+    def initialize(name, class_context, event_registry, **opts)
+      @name = name
+      @class_context = class_context
+      @options = opts
+
+      @states = NxtStateMachine::StateRegistry.new
+      @transitions = []
+      @events = event_registry
+      @callbacks = CallbackRegistry.new
+      @error_callback_registry = ErrorCallbackRegistry.new
+      @defuse_registry = DefuseRegistry.new
```

Comment:
It's not easily obvious what a `DefuseRegistry` is, and it's not explained anywhere.